super_considered_obligatory.py

```diff
@@ -0,0 +1,130 @@
+"""We were discussing when to invoke `super().tearDown` on Django tests... a friend said
+>  a built-in context manager type thing for super calls would be a cool language feature
+
+The features below impliment that, go on, use them in real tests
+your future self will be ~proud~ ~confused~ ... cursing you
+"""
+import functools
+import inspect
+
+
+def super_later(f):
+    """Makes sure the wrapped function tries to call its
+    namesake on super() as the last thing it does
+    """
+
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        fs_cls = context_that_defined(f)
+        fs_cls_parent = inspect.getmro(fs_cls)[1]
+        fs_cls_parent_f = getattr(fs_cls_parent, f.__name__, None)
+        try:
+            f(*args, **kwargs)
+        except Exception as err:
+            raise (err)
+        finally:
+            if fs_cls_parent_f:
+                fs_cls_parent_f(*args, **kwargs)
+
+    return wrapper
+
+
+def context_that_defined(callable):
+    """Try to get the context where `callable` was defined
+    """
+    if inspect.ismethod(callable):
+        for cls in inspect.getmro(callable.__self__.__class__):
+            if cls.__dict__.get(callable.__name__) is callable:
+                return cls
+        # 1st failover parse __qualname__
+        callable = callable.__func__
+    if inspect.isfunction(callable):
+        cls = getattr(
+            inspect.getmodule(callable),
+            callable.__qualname__.split(".<locals>", 1)[0].rsplit(".", 1)[0],
+        )
+        if isinstance(cls, type):
+            return cls
+    # 2nd failover for descriptors
+    return getattr(callable, "__objclass__", None)
+
+
+class DecoratorParent:
+    def some_method(self):
+        print("called Parent.some_method")
+
+
+class DecoratorChild(DecoratorParent):
+    @super_later
+    def some_method(self):
+        print("called Child.some_method")
+
+
+class FinalSuperMeta(type):
+    def __new__(cls, name, bases, local):
+        for attr in local:
+            value = local[attr]
+            if callable(value):
+                local[attr] = super_later(value)
+        return type.__new__(cls, name, bases, local)
+
+
+class NervousParent(metaclass=FinalSuperMeta):
+    """All child classes will attempt
+    to call super on _all_ their methods
+    """
+
+    def spam(self):
+        print("called NervousParent.spam")
+
+
+class ObedientChild(NervousParent):
```

Comment:
y'know the more read naming convention
 the less and less keen I am on the 
Parent nagging their Child to call them 
metaphore

especially if we're using words like 'force' in the docs...

suggestions for alternatives _very_ welcome